tests/chat/mlx/test_chat_template.py

```
@@ -213,7 +212,131 @@ def test_kwargs_passthrough(self):
         assert isinstance(prompt, str)
+
+    def test_tool_calls_json_conversion(self):
+        """Test that tool_calls JSON arguments are converted to dicts for Jinja template"""
+        # This is a unit test for the JSON conversion logic in chat_template.py
+        # We test the conversion logic directly without loading models
+
+        from mlx_omni_server.chat.mlx.tools.chat_template import ChatTemplate
+
+        # Mock tokenizer (we don't need actual tokenization for this test)
+        class MockTokenizer:
+            def apply_chat_template(self, conversation, **kwargs):
```

Ruff (0.14.8)

Unused method argument: `kwargs`

(ARG002)